esrally/metrics.py

```diff
@@ -250,7 +251,22 @@ def results_template(self):
 
     def _read(self, template_name):
         with open("%s/resources/%s.json" % (self.script_dir, template_name), encoding="utf-8") as f:
-            return f.read()
+            # if we're using a persistent metrics store, then we should check and update the index templates settings
+            if self.cfg.opts("reporting", "datastore.type") == "elasticsearch":
```

Comment:
This class doesn't get called when there is an in-memory metrics store; the references are always from EsXYZStore i.e.  https://github.com/elastic/rally/blob/ede9670d50f1d0330baf09738a01e6535b17a49f/esrally/metrics.py#L862 / https://github.com/elastic/rally/blob/ede9670d50f1d0330baf09738a01e6535b17a49f/esrally/metrics.py#L1607 / https://github.com/elastic/rally/blob/ede9670d50f1d0330baf09738a01e6535b17a49f/esrally/metrics.py#L1700

So we don't need this conditional. Therefore, we can also move the code inside `_update_template()` here and reference the shard settings from instance variables, see the comment below.

Reply:
Good pick up!